File: python/test_fetch_akahu.py
import os
from dotenv import load_dotenv
from akahu_api import AkahuAPI

def test_akahu_fetch_transactions():
    # Load environment variables from .env file
    load_dotenv()

    # Fetch Akahu credentials from environment variables
    AKAHU_APP_TOKEN = os.getenv("AKAHU_APP_TOKEN")
    AKAHU_USER_TOKEN = os.getenv("AKAHU_USER_TOKEN")
    AKAHU_ACCOUNT_ID = os.getenv("AKAHU_ACCOUNT_ID")

    # Ensure required environment variables are set
    if not all([AKAHU_APP_TOKEN, AKAHU_USER_TOKEN, AKAHU_ACCOUNT_ID]):
        print("Error: Missing one or more required environment variables for Akahu API.")
        return

    # Initialize Akahu API
    akahu_api = AkahuAPI(AKAHU_APP_TOKEN, AKAHU_USER_TOKEN)

    # Fetch transactions for a specific account
    try:
        transactions = akahu_api.get_transactions(AKAHU_ACCOUNT_ID)
        print(f"Fetched {len(transactions.get('items', []))} transactions for account ID {AKAHU_ACCOUNT_ID}.")
        for transaction in transactions.get('items', []):
            print(transaction)
    except RuntimeError as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    test_akahu_fetch_transactions()

File: test-fetch-akahu.js
const { AkahuClient } = require('akahu');
const dotenv = require('dotenv');
const path = require('path');

// Load environment variables from .env file
dotenv.config({ path: path.join(__dirname, '.env') });

// Akahu API credentials
const appToken = process.env.AKAHU_APP_TOKEN;
const userToken = process.env.AKAHU_USER_TOKEN;

// Akahu account ID to fetch transactions for
const accountId = 'acc_clp77cei7000308l4ed6be3i9';

// Initialize Akahu client
const akahuClient = new AkahuClient({
    appToken: appToken,
});

async function fetchTransactionsForAccount(accountId) {
    try {
        console.log(`Fetching transactions for account: ${accountId}`);

        // Fetch transactions for the specific account using listTransactions
        const response = await akahuClient.accounts.listTransactions(userToken, accountId);

        console.log(`Fetched ${response.items.length} transactions for account: ${accountId}`);
        response.items.forEach((transaction, index) => {
            console.log(`${index + 1}: ${transaction.date} - ${transaction.description} - $${transaction.amount}`);
        });
    } catch (error) {
        console.error('Error fetching transactions:', error.message);
    }
}

// Call the function to fetch transactions
fetchTransactionsForAccount(accountId);
